### PR TITLE
Add LiveChat icon to WPAdmin on Atomic only.

### PR DESCRIPTION
**Pull request by getdave**
Adds the "FAB" LiveChat icon to WPAdmin on WordPress.com envs only. This is a non-interactive element as it is not yet possible to "embed" the actual FAB widget from Calypso into WPAdmin. This acts as a temporary workaround to allow the user access to "help" via a direct link to `wordpress.com/help`.

Screen Shot 2021-02-22 at 11 37 17

To do this we must:

* Include the same markup as per Calypso FAB.
* Inline the Grid Icon SVG so as not to rely on Gridicons.
* Apply layout styles based on Calypso styles.

In addition we must use the existing color schemes unification process to define custom color overides for the icon depending on which color scheme is being used in WPAdmin. This involves adding new vars to each color scheme file and then applying these to the FAB styling.

Fixes https://github.com/Automattic/wp-calypso/issues/50271
Fixes https://github.com/Automattic/wp-calypso/issues/50367

#### Please note

* I discussed this with @jeherve and we decided putting in Masterbar was an ok choice - at least for the sake of expediency for now.
* We are going to ignore custom color schemes for the LiveChat icon for now. We can follow up to add color scheme support later as required.

## Changes proposed in this Pull Request:
* Add LiveChat icon to WPAdmin for Atomic only.

## Jetpack product discussion

See paYJgx-1nM-p2

#### Does this pull request change what data or activity we track or use?
No

## Testing instructions:

The following steps should be taken on both Simple & Atomic (the visual icon check should be tested across a variety of screen sizes):

* Go to WPAdmin. 
* See LiveChat icon in bottom right.
* Click icon to be taken to `wordpress.com/help`.
* Go to `wordpress.com` and open your site in _Calypso_.
* Create a new Post and check you are within the Gutenframed editor (here's how you check for that PCYsg-lwV-p2).
* See only a _single_ icon in bottom right - it should be the proper Calypso FAB that you can interact with and not the static version.
* Go back to WPAdmin and create a new post **_not_** using the Gutenframed editor. 
* Again you should see a single icon in the bottom right but it should be the static version that is just a link and not the interactive Calypso version.

#### Color Schemes
* Also try changing your color scheme and then re-testing to ensure the icon picks up the color schemes correctly.

### Atomic sites

* Get Atomic site with JP Beta.
* Checkout this branch using JP Beta.
* Follow testing steps above.

### Simple sites

This doesn't work yet because we need to update JP Fusion to sync the `inline-help` directory within masterbar to Dotcom Simple codebase.

Once this PR is provisionally accepted for Atomic sites then we'll merge this diff (D57403-code) to update JP sync and then rebase all the things.

* Follow testing steps above.

## Proposed changelog entry for your changes:

* Adds WP.com Live Chat icon (link only) to bottom right of WPAdmin for Atomic sites only.

### CODE REVIEW

**Comment by getdave** [comment]
@frontdevde I'd really appreciate your help here with:

* Getting the box shadow on the icon to match up with Calypso - for some reason the var is ignored.
* Getting the sizing of the icon the same as Calypso - it isn't matching up. Must be missing some key CSS rules.

**Comment by frontdevde** [comment]
>@frontdevde I'd really appreciate your help here with:
Getting the box shadow on the icon to match up with Calypso - for some reason the var is ignored.
Getting the sizing of the icon the same as Calypso - it isn't matching up. Must be missing some key CSS rules.

Pushed a couple of updates, the issues were related to
1) the var needed the actual individual r/g/b values
2) the box model was missing (in Calypso it's inherited from the button component)

Hope that helps!

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:35]: Actions don't need to return anything, that's something only required by filters.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:40]: The "Help" title here should be translated.

**Review by jeherve — CHANGES_REQUESTED** [review]
You'll need to fix those linting issues before we can merge this.

**Review by mmtr — CHANGES_REQUESTED** [review]
Nice work so far, @getdave. I only took a first quick look without testing it, but I think this is going in the right direction.

My only concern is that this WP Admin FAB would conflict with the Calypso FAB on screens that we embed through an iframe (e.g. block editor / Gutenframe). It may be good to check the non-presence of the `frame-nonce` GET param before injecting the WP Admin FAB, since that params indicated the screen is going to be embedded on a iframe.

It'd be also good to solve all the linting warnings/errors.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:35]: Whoops! Left this in as typo. Thanks for catching.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php]: Done

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php]: Done

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:42]: We can do a more specific target by looking for the `frame-nonce` GET param instead of checking that `framce-nonce` is present on the URL.

Also, the presence of the `frame-nonce` GET param does not necessarily indicates a Gutenframe context, since it can be used in other places (e.g. [post preview](https://github.com/Automattic/jetpack/blob/d03c9e834b52ba5a489ad2fd991f11f72f119f7e/projects/plugins/jetpack/class.frame-nonce-preview.php#L44)). Due to that, I think it's better to don't use any reference to Gutenframe in the variables.

```suggestion
		if ( ! empty( $_GET['frame-nonce'] ) {
			return;
		}
```

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php]: Good points.

**Comment by getdave** [comment]
@mmtr I fixed the doubling up of the icons in Gutenframe. 

@frontdevde Just need to tidy up the CSS to move things to Sass and avoid CSS vars.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:80]: Could we use a more specific name to avoid potential conflicts with any other plugins that may style and / or add `inline-help` containers to wp-admin?

File: projects/plugins/jetpack/modules/masterbar/inline-help/inline-help.css
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/inline-help/inline-help.css:2]: You'll want to use tabs instead of spaces in all that file, to avoid folks accidentally editing all lines when they'll edit this file in their editor in the future.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:35]: Should we insert the icon at all in the block editor?

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:35]: I did this because that matches up with what happens in Calypo Gutenframe. There is a FAB on Calypso editor.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:80]: I agree. This is wide open. Lets try `.a8c-faux-inline-help`?

Comment (jeherve) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:80]: Sounds good!

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php]: Done

File: projects/plugins/jetpack/modules/masterbar/inline-help/inline-help.css
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/inline-help.css]: I've used my editor to convert to tabs. Looks better?

**Comment by getdave** [comment]
@jeherve We've also been asked to add tracking when someone clicks the link. Do we need to land that in this PR to hit Atomic in time for release? 

https://github.com/Automattic/wp-calypso/issues/50367

**Comment by jeherve** [comment]
> Do we need to land that in this PR to hit Atomic in time for release or can we follow up?

Let's add it to this PR if you can.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:13]: We'll have to fix those linting errors before we are able to merge this PR.

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:13]: Argh more linting. I'll fix asap.

File: projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:9]:
```suggestion
	<a data-jptracks-name="<?php echo esc_attr( $args['tracking_event_name'] ); ?>" class="<?php echo esc_attr( $jp_tracking_classname ); ?> a8c-faux-inline-help__button" href="<?php echo esc_url( $args['href'] ); ?>" target="_blank" rel="noopener noreferrer" title="<?php echo esc_attr__( 'Help', 'jetpack' ); ?>">
```

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php]: Done

File: projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:5]: Any reason for keeping this on a separate variable instead of directly hard-coding it in the HTLM below?

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:5]: So I could easily comment it to make sure it's obvious that it's required for tracking purposes and not for styling.

So many times in my career have I seen people remove such classes or accidentally break them. Adding this way makes it self documenting and harder to break.

Personal preference though...

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:5]: Incidentally, this is why I prefer `data-` attributes for such things as opposed to classes.

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:5]: Why do we even need the classname if we have a `data-jptracks-` which we could use to select things in the DOM with a minimal perf penalty?

File: projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:9]: Not use if the tracks setup is conflicting with the link, but clicking on the button does not take me to wordpress.com/help, although I can see the `href` attribute properly set:

Feb-23-2021 14-41-36

File: projects/plugins/jetpack/modules/masterbar/inline-help/inline-help.css
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/inline-help.css:54]: Turns out the problem I'm facing in https://github.com/Automattic/jetpack/pull/18885/files#r581043422 is caused by the SVG of the gridicon capturing the click. We can handle this by making sure the gridicon ignore any click event:

```suggestion
.a8c-faux-inline-help .a8c-faux-inline-help__button .gridicon {
	pointer-events: none;
```

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/inline-help.css:54]: Classic bug. Thanks for spotting.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php]: Fixed

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:89]: [`load_template` is not very used in Jetpack](https://github.com/Automattic/jetpack/search?q=load_template), and I noted there is a [preference for using `printf`](https://github.com/Automattic/jetpack/search?q=printf) instead. Do you have any strong opinion against it?
```suggestion
		printf(
			'<div class="a8c-faux-inline-help"><a href="%1$s" target="_blank" rel="noopener noreferrer" title="%2$s" class="a8c-faux-inline-help__button jptracks" data-jptracks-name="%3$s">%4$s</a></div>',
			esc_url( 'https://wordpress.com/help' ),
			esc_attr__( 'Help', 'jetpack' ),
			esc_attr( 'jetpack_masterbar_inline_help' ),
			wp_kses( $gridicon_help, $svg_allowed )
		);
```

That should also get rid of the necessity of disabling the `WordPress.Security.EscapeOutput.OutputNotEscaped` rule.

**Review by mmtr — COMMENTED** [review]
Nice work, @getdave! This works quite well for me. Also verified the FAB uses the correct colors according to the current color scheme selected.

I left a minor question regarding the template used for loading the inline help markup.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:89]: Personally, I find it much easier to maintain HTML in a dedicated file and simply drop into PHP when I need to. 

Indeed, that's what WordPress itself does 90% of the time when dealing with markup.

There is a convention in WP land where people like to manipulate strings of markup within PHP. I would personally avoid this at all costs but if it's not the JP way then that's fine. 

> WordPress.Security.EscapeOutput.OutputNotEscaped

This is a tool - sometimes it's "wrong" 😄

**Comment by mmtr** [comment]
Ah, forgot to note that I also checked the tracks event is triggered correctly:

Screen Shot 2021-02-23 at 16 16 38

**Comment by getdave** [comment]
> Ah, forgot to note that I also checked the tracks event is triggered correctly:
> 
> Screen Shot 2021-02-23 at 16 16 38

I couldn't get this to trigger locally for me. How did you manage it?

**Comment by getdave** [comment]
> Nice work, @getdave! This works quite well for me. Also verified the FAB uses the correct colors according to the current color scheme selected.

Kudos to @frontdevde for this.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:89]: > I would personally avoid this at all costs but if it's not the JP way then that's fine.

I'm going to defer to @jeherve on this one, since I'm not sure if there is any guideline regarding this (I just observed that preference by looking at the existing code).

**Comment by mmtr** [comment]
> > Ah, forgot to note that I also checked the tracks event is triggered correctly:
> > Screen Shot 2021-02-23 at 16 16 38
> 
> I couldn't get this to trigger locally for me. How did you manage it?

I ran `localStorage.setItem( 'debug', 'dops:analytics' );` in the JS console and reloaded the page. All tracked events showed up after that in the console.

**Comment by getdave** [comment]
At the moment this is blocked by me not being sure of the best way to get the diff to be in sync with these changes.

**Comment by mmtr** [comment]
> At the moment this is blocked by me not being sure of the best way to get the diff to be in sync with these changes.

You should be able to commandeer D57354-code and update the diff after manually copy/paste the `inline-help` folder.

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:89]: Happy either way. If it avoids problems I'm fine to covert it back.

**Comment by getdave** [comment]
Waiting on review of accompanying diff which covers Simple sites.

D57354-code

File: projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:8]: This is no longer needed, right?

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/inline-help/inline-help-template.php:8]: It is needed for `$args`.

**Review by mmtr — APPROVED** [review]
LGTM! :shipit:

File: projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php:91]: I would personally have `printf` things right here, without the extra template; that would have avoided having to add those phpcs exceptions. But that's more of a personal preference, it's fine like this I think.

**Review by jeherve — APPROVED** [review]
This looks good to me. 👍

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.5` in 2e0f703ee107166df4a2dfec1c2a94fc662f91dc

**Pull request merged**
